src/forcing/ForcingsEngineLumpedDataProvider.cpp

```diff
@@ -0,0 +1,144 @@
+#include "DataProvider.hpp"
+#include <forcing/ForcingsEngineLumpedDataProvider.hpp>
+
+namespace data_access {
+
+using Provider     = ForcingsEngineLumpedDataProvider;
+using BaseProvider = Provider::base_type;
+
+std::size_t convert_divide_id_stoi(const std::string& divide_id)
+{
+    auto separator = divide_id.find('-');
+    const char* split = (
+        separator == std::string::npos
+        ? &divide_id[0]
+        : &divide_id[separator + 1]
+    );
+
+    return std::atol(split);
+}
+
+Provider::ForcingsEngineLumpedDataProvider(
+    const std::string& init,
+    std::size_t time_begin_seconds,
+    std::size_t time_end_seconds,
+    const std::string& divide_id
+)
+  : BaseProvider(init, time_begin_seconds, time_end_seconds)
+{
+    divide_id_ = convert_divide_id_stoi(divide_id);
+
+    // Check that CAT-ID is an available output name, otherwise we most likely aren't
+    // running the correct configuration of the forcings engine for this class.
+    const auto cat_id_pos = std::find(var_output_names_.begin(), var_output_names_.end(), "CAT-ID");
+    if (cat_id_pos == var_output_names_.end()) {
+        throw std::runtime_error{
+            "Failed to initialize ForcingsEngineLumpedDataProvider: `CAT-ID` is not an output variable of the forcings engine."
+            " Is it running with `GRID_TYPE` set to 'hydrofabric'?"
```

Comment:
For this error message and the one below, it would be good to include the value of `init` so it's easy to know where to look for the misconfiguration.